tests/helper/tests/sshd.py

```diff
@@ -0,0 +1,139 @@
+        if cls.failed_before:
+            raise Exception("This test failed before in another feature")
+
+        (enabled_features, my_feature) = features
+
+        # check if test is disabled in a feature
+        test_is_disabled = utils.disabled_by(
+            enabled_features, 'blacklisted_packages')
+        if not len(test_is_disabled) == 0:
+            raise DisabledBy(
+                "Test is explicitly disabled by features " +
+                    f"{', '.join(test_is_disabled)}")
+
+        # check if the test is part of the features used to build the
+        # gardenlinux image
+        if my_feature not in enabled_features:
+            raise NotPartOfFeatureError(
+                f"Feature {my_feature} this test belongs to is not enabled")
+
+        # first check if there is already an instance of this class, if it is
+        # the first time this instance is initiated add the class variable
+        # instance containing the instance itself and then do the actual
+        # testing.
+        if not hasattr(cls, 'instance'):
+            cls.instance = super(Sshd, cls).__new__(cls)
+
+            (exit_code, output, error) = client.execute_command("sshd -T",
+                                                                quiet=True)
+            assert exit_code == 0, f"no {error=} expected"
+
+            sshd_config = cls._create_dict(cls.instance, output)
+
+            expected = utils.read_test_config(
+                enabled_features, 'sshd', '_expected')
+            sshd_expected = cls._create_dict(cls.instance, expected)
+
+            for key, value in sshd_config.items():
+                if key in sshd_expected:
+                    sshd_config_value = cls._normalize_value(
+                                            cls.instance, value)
+                    expected_value = cls._normalize_value(
+                                        cls.instance, sshd_expected[key])
+                    missing, where = cls._compare_as_set(cls.instance,
+                                        sshd_config_value, expected_value)
+                    if where == cls.NOT_IN_SSHD_CONFIG:
+                        cls.failed_before = True
+                        raise TestFailed(f"For {key} '{', '.join(missing)}'" +
+                            " is expected but missing")
+                    if where == cls.IN_SSHD_CONFIG:
+                        cls.failed_before = True
+                        raise TestFailed(f"For {key} '{', '.join(missing)}'" +
+                            " is set but not expected")
+                    if where == cls.DIFFERENT:
+                        cls.failed_before = True
+                        raise TestFailed(f"For {key} " +
+                            f"'{', '.join(expected_value)}' is expected, " +
+                            f"'{', '.join(sshd_config_value)}' is set")
+
+        return cls.instance
+
+    def _create_dict(self, input):
```

Comment:
If we define these methods as functions of this python module, we don't need to call them with `cls.instance` each time. Since they do not process the class object or any attribute of it, we don't need to pass this information.